### ISSUE TITLE
tensorboard - visualizing pre-trained embeddings - projector shows up in the inactive tab

### ISSUE THREAD

**Issue by zotroneneis** [tensorflow/tensorboard#718]
I am trying to visualize pre-trained word embeddings. I have tried the code posted [here](https://stackoverflow.com/questions/41258391/tensorboard-embedding-example) and I have also looked at the [GitHub issue](https://github.com/tensorflow/tensorflow/issues/6322) and [related posts](https://stackoverflow.com/questions/40849116/how-to-use-tensorboard-embedding-projector). 

The visualization is neither working when including the visualization in the rest of my model code, nor when separating it into a different file. Every time, the "projector" shows up under the "inactive" tab.

I would like to visualize the embeddings together with the rest of my model summaries.
The relevant code is here:
        
        [...]
        embedding_file = os.path.join(HOME, self.config['general']['embedding_file'])
        with open(embedding_file, 'rb') as f:
            self.embds = pickle.load(f)

        self.vocab_size, self.embd_size = self.embds.shape
        self.graph = self.build_graph(tf.Graph())

        with self.graph.as_default():
            self.saver = tf.train.Saver(
                max_to_keep=5, )
            self.init_op = tf.global_variables_initializer()

            sess_config = tf.ConfigProto()
            self.sess = tf.Session(config=sess_config, graph=self.graph)

            self.sess.run(self.init_op)
            self.sess.run(self.embed_init, feed_dict={self.embed_placeholder: self.embds})

            self.sw_train = tf.summary.FileWriter(self.tb_dir+'/train', self.graph)
            self.sw_test = tf.summary.FileWriter(self.tb_dir+'/test')

            self.projector_config = tf.contrib.tensorboard.plugins.projector.ProjectorConfig()
            self.embedding_conf = self.projector_config.embeddings.add()
            self.embedding_conf.tensor_name = self.embedding_matrix.name
            train_tb = self.tb_dir + '/train/'
            self.embedding_conf.metadata_path = train_tb + 'vocab.tsv'
            tf.contrib.tensorboard.plugins.projector.visualize_embeddings(self.sw_train, self.projector_config)

            self.saver.save(self.sess, self.tb_dir + '/train/model.ckpt')

    def build_graph(self, graph):
        with graph.as_default():
            tf.set_random_seed(self.random_seed)
            with tf.variable_scope('embedding'):
                self.embedding_matrix = tf.Variable(tf.constant(0.0, shape=[self.vocab_size, self.embd_size]), trainable=False, name='embedding')
                self.embed_placeholder = tf.placeholder(tf.float32, [self.vocab_size, self.embd_size])
                self.embed_init = self.embedding_matrix.assign(self.embed_placeholder)
                [...]

TensorFlow version: 1.4.0

**Comment by zotroneneis** [comment]
[tensorboard.zip](https://github.com/tensorflow/tensorboard/files/1460735/tensorboard.zip)

**Comment by vitalyli** [comment]
This works for me:

    tf.reset_default_graph()
    sess = tf.InteractiveSession()
    X = tf.Variable([0.0], name='embedding')
    place = tf.placeholder(tf.float32, shape=[embDic.vocab_size, embDic.dim])
    set_x = tf.assign(X, place, validate_shape=False)

    sess.run(tf.global_variables_initializer())
    sess.run(set_x, feed_dict={place: embDic.embeddings})

    # write labels
    with open(FLAGS.out_dir + '/metadata.tsv', 'w') as f:
        for word in embDic.vocab:
            line_out = "%s\n" % word
            f.write(line_out.encode('utf-8'))

    # create a TensorFlow summary writer
    summary_writer = tf.summary.FileWriter(FLAGS.out_dir + '/' + 'emb_viz_%s.log' % version, sess.graph)
    config = projector.ProjectorConfig()
    embedding_conf = config.embeddings.add()
    embedding_conf.tensor_name = 'embedding:0'
    embedding_conf.metadata_path = os.path.join(FLAGS.out_dir, 'metadata.tsv')
    projector.visualize_embeddings(summary_writer, config)

    saver = tf.train.Saver()
    saver.save(sess, os.path.join(FLAGS.out_dir, "model.ckpt"))

**Comment by jart** [comment]
Thanks for helping our friend @vitalyli.

**Comment by zotroneneis** [comment]
Although that's very similar to what I did already I have tried your code. Regrettable, the problem still remains the same.

**Comment by Cory-M** [comment]
Same thing happens to me. It works well on local computer but doesn't work on the cluster. Have you figured out what's wrong?
